Source/Eto.Test/Eto.Test/Sections/Controls/DrawableSection.cs

```diff
@@ -95,11 +130,162 @@ Control LargeCanvas()
+							Content = new TableLayout(
+								true,
+								null,
+								TableLayout.HorizontalScaled(
+									null,
+									new Panel
+									{
+										BackgroundColor = Color.FromArgb(255, 0, 0),
+										Size = new Size(10, 10)
+									}
+								)
+							)
+						},
+						new Drawable
+						{
+							BackgroundColor = Color.FromArgb(0, 255, 0, 128),
+							Content = new TableLayout(
+								true,
+								null,
+								TableLayout.HorizontalScaled(
+									new Panel
+									{
+										BackgroundColor = Color.FromArgb(0, 255, 0),
+										Size = new Size(10, 10)
+									},
+									null
+								)
+							)
+						}
+					),
+					TableLayout.HorizontalScaled
+					(
+						new Drawable
+						{
+							BackgroundColor = Color.FromArgb(0, 0, 255, 128),
+							Content = new TableLayout(
+								true,
+								TableLayout.HorizontalScaled(
+									null,
+									new Panel
+									{
+										BackgroundColor = Color.FromArgb(0, 0, 255),
+										Size = new Size(10, 10)
+									}
+								),
+								null
+							)
+						},
+						new Drawable
+						{
+							BackgroundColor = Color.FromArgb(0, 0, 0, 128),
+							Content = new TableLayout(
+								true,
+								TableLayout.HorizontalScaled(
+									new Panel
+									{
+										BackgroundColor = Color.FromArgb(0, 0, 0),
+										Size = new Size(10, 10)
+									},
+									null
+								),
+								null
+							),
+						}
+					)
+				)
+			}.With(it => it.Paint += (s,pe) =>
+			{
+				using (var p = new Pen(Colors.Black, 3f))
+				{
+					for(int i = 4, n = Math.Max(it.Width, it.Height); i < n; i += 8)
+					{
+						pe.Graphics.DrawLine(p, i, 0, i + n, n);
+						pe.Graphics.DrawLine(p, 0, i, n, i + n);
+					}
+				}
+			});
+		}
+
+		Control Tools(int n)
+		{
+			var stack = new StackLayout
+			{
+				BackgroundColor = Colors.Transparent,
+				Orientation = Orientation.Horizontal,
+				VerticalContentAlignment = VerticalAlignment.Center,
+				Padding = 4,
+				Spacing = 4,
+			};
+			if (n > 0)
+			{
+				stack.Items.Add("Label:");
+				for (int i = 0; i < n; i++)
+				{
+					stack.Items.Add(new TextBox
+					{
+						Text = "Box" + ( i + 1 )
+					});
+				}
+			}
+			var control = new Drawable
+			{
+				Content = stack
+			};
+			control.Paint += (s, pe) =>
+			{
+				int w = control.Width;
+				int h = control.Height;
+				var c1 = Color.FromGrayscale(0.8f);
+				var c2 = Color.FromGrayscale(0.6f);
+				using (var b = new LinearGradientBrush(
+					c1, c2, new PointF(1, 0), new PointF(1, h)))
+					pe.Graphics.FillRectangle(b, 1, 0, w-2, h);
+				pe.Graphics.DrawLine(c1, 0, 0, 0, h);
+				pe.Graphics.DrawLine(c2, w-1, 0, w-1, h);
+			};
+			return control;
+		}
+
+		void LogEvents(Drawable control, string name)
 		{
 			control.Paint += delegate(object sender, PaintEventArgs pe)
 			{
-				Log.Write(control, "Paint, ClipRectangle: {0}", pe.ClipRectangle);
+				Log.Write(name, "Paint, ClipRectangle: {0}", pe.ClipRectangle);
```

Comment:
The log now contains name of the drawable, to separate the problems when more events are fired then expected.